bundles/org.openhab.binding.zm/src/main/java/org/openhab/binding/zm/internal/handler/ZmBridgeHandler.java

```diff
@@ -0,0 +1,645 @@
+                    }
+                } catch (NumberFormatException e) {
+                    logger.warn("Bridge: Badly formatted version number {}", softwareVersion);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Badly formatted version number");
+                }
+            } else {
+                logger.warn("Bridge: Can't parse software version {}", softwareVersion);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Can't parse software version");
+            }
+        } else {
+            logger.warn("Bridge: Unable to determine Zoneminder software version");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Software version is null");
+        }
+        return false;
+    }
+
+    private boolean checkApiVersion(@Nullable String apiVersion) {
+        logger.debug("Bridge: Zoneminder API version is {}", apiVersion);
+        if (apiVersion != null) {
+            String[] versionParts = apiVersion.split("\\.");
+            if (versionParts.length >= 2) {
+                try {
+                    int versionMajor = Integer.parseInt(versionParts[0]);
+                    if (versionMajor >= 2) {
+                        logger.debug("Bridge: Zoneminder API version check OK");
+                        return true;
+                    } else {
+                        logger.warn("Bridge: Requires API version >= 2.0. This Zoneminder is version {}", apiVersion);
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                "Requires API version >= 2.0");
+                    }
+                } catch (NumberFormatException e) {
+                    logger.warn("Bridge: Badly formatted version number {}", apiVersion);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Badly formatted API version number");
+                }
+            } else {
+                logger.warn("Bridge: Can't parse software version {}", apiVersion);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Can't parse API version");
+            }
+        } else {
+            logger.warn("Bridge: Unable to determine Zoneminder API version");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "API version is null");
+        }
+        return false;
+    }
+
+    /*
+     * The refresh job is executed every second
+     * - updates the monitor handlers every monitorsInterval seconds, and
+     * - runs the monitor discovery every discoveryInterval seconds
+     */
+    private void refresh() {
+        try {
+            refreshMonitors();
+            discoverMonitors();
+        } catch (RuntimeException e) {
```

Comment:
What type of exception do you expect here? Can you specify the concrete type?

Reply:
I'm trying to prevent the refresh job from silently failing due to an unexpected exception. The exception normally would result from a poorly formatted response from the Zoneminder server, but the exact exception type is hard to predict. If not caught, the refresh job will be canceled and nothing will be logged. Catching the runtime exception is here to prevent that situation.

Reply:
You already catch the JsonSyntaxException and IOExceptions. That should cover most unexpected cases. However, uncatched unchecked exceptions like `RuntimeException` are logged by the framework. They don't need to be logged by the binding.

Reply:
Understood. However, I'm pretty sure I added this because I got an exception (I don't remember which one) that caused the refresh job to be canceled and was not logged by the framework. I can remove this, then revisit if the problem reoccurs.

Reply:
If it wasn't logged, it might be a bug in the framework. Would be good to remove it and investigate that.

Reply:
Ok, I removed it here and one other place.